### PR TITLE
Remove the class-depth and holder hacks.

### PR DESCRIPTION
**Pull request by floitsch**
Rewrite the inheritance computation making it work better with mixins.

### CODE REVIEW

File: tools/toitdoc/src/inheritance.toit [hunk 1]
```diff
@@ -234,81 +230,131 @@ class InheritanceBuilder:
...
+  compute-inherited-for klass/lsp.Class --superclass/lsp.Class --super-inherited/List -> List:
+    class-shaped := ShapedMap
 
-    inherited[klass] = class-inherited
+    (klass.methods + klass.fields).do: | entry/lsp.ClassMember |
+      // We initialize the inheritance order to a high value, so that it is
+      // always higher than the inheritance order of the super class or mixins.
+      // The current class members are discarded at the end of the function, so
+      // don't need any fixing up.
+      member := Member entry --inheritance-order=10000000
```
Comment (kasperl) [inline tools/toitdoc/src/inheritance.toit:294]:
```suggestion
      member := Member entry --inheritance-order=10_000_000
```

File: tools/toitdoc/src/shadow.toit [hunk 2]
```diff
@@ -439,10 +435,11 @@ compute-override-phase-positional super-iter/NamedIterator -> bool
     for i := from; i <= to; i++:
       if arities[i] <= 0:
         // Either another old overrider, or not overridden yet.
-        class-depth := -hack-class-depths[hack-holders[old-member.as-toit-member]]
-        if class-depth < arities[i]:
+        // Note that we use the negative of the inheritance order as discussed above.
+        inheritance-order := -old-member.inheritance-order
```
Comment (kasperl) [inline tools/toitdoc/src/shadow.toit:439]: Maybe give this a different name? Once negated, it is no longer the inheritance order.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/toitdoc/src/inheritance.toit (hunk 1 above)
Comment (floitsch) [reply tools/toitdoc/src/inheritance.toit]: done.

File: tools/toitdoc/src/shadow.toit (hunk 2 above)
Comment (floitsch) [reply tools/toitdoc/src/shadow.toit]: done.

**Pull request merged**